src/fpm_targets.f90

```diff
@@ -249,30 +274,87 @@ subroutine build_target_list(targets,model)
 
     contains
 
+    function get_file_scope_flags(source, profile) result(file_scope_flag)
+        ! Try to match source%file_name in profile%file_scope_flags
+        !
+        !
+        type(srcfile_t), intent(in) :: source
+        type(profile_config_t), intent(in) :: profile
+
+        character(:), allocatable :: file_scope_flag, current
+        integer :: i
+
+        file_scope_flag = ""
+
+        if (allocated(profile%file_scope_flags)) then
+            associate(fflags=>profile%file_scope_flags)
+                do i=1,size(fflags)
+                    if (source%file_name.eq.fflags(i)%file_name) then
+                        file_scope_flag = fflags(i)%flags//" "
+                    end if
+                end do
+            end associate
+        end if
+    end function get_file_scope_flags
+
     function get_object_name(source) result(object_file)
         ! Generate object target path from source name and model params
         !
         !
-        type(srcfile_t), intent(in) :: source
+        type(srcfile_t), intent(inout) :: source
```

Comment:
Again, we don't want to be modifying the model at this point. We may need to think how we can move this code to occur before we get to `build_target_list` - _i.e._ sort out all the per-source flags when we build the model.